graphql_compiler/schema_transformation/rename_schema.py

```diff
@@ -186,25 +238,28 @@ class RenameSchemaTypesVisitor(Visitor):
         }
     )
 
-    def __init__(self, renamings, query_type, scalar_types):
+    def __init__(
+        self, renamings: Dict[str, Optional[str]], query_type: str, scalar_types: Set[str]
+    ) -> None:
         """Create a visitor for renaming types in a schema AST.
 
         Args:
-            renamings: Dict[str, str], mapping from original type name to renamed type name.
-                       Any name not in the dict will be unchanged
+            renamings: Dict[str, Optional[str]], mapping from original type name to renamed type
+                       name or None (for type suppression). Any name not in the dict will be
+                       unchanged
             query_type: str, name of the query type (e.g. RootSchemaQuery), which will not
                         be renamed
             scalar_types: Set[str], set of names of all scalars used in the schema, including
                           all user defined scalars and any builtin scalars that were used
         """
         self.renamings = renamings
-        self.reverse_name_map = {}  # Dict[str, str], from renamed type name to original type name
-        # reverse_name_map contains all types, including those that were unchanged
+        self.reverse_name_map: Dict[str, str] = {}  # From renamed type name to original type name
+        # reverse_name_map contains all non-suppressed types, including those that were unchanged
         self.query_type = query_type
         self.scalar_types = frozenset(scalar_types)
         self.builtin_types = frozenset({"String", "Int", "Float", "Boolean", "ID"})
 
-    def _rename_name_and_add_to_record(self, node):
+    def _rename_name_and_add_to_record(self, node: Node) -> Union[type(REMOVE), Optional[Node]]:
```

Comment:
This is the other lint error we're seeing, which is because the [enter](https://github.com/kensho-technologies/graphql-compiler/pull/834/files#diff-3b859f46e97b0af834db0b3b7de08658R321) function has `node` type as `Node`, but this function will only be called with a node of type in `self.rename_types`, and all of those nodes have names. Not sure how to write that in the type-hint though-- putting anything except `Node` as the type-hint makes mypy complain in the `enter` function that calls `_rename_name_and_add_to_record`

Reply:
I've confirmed that the following workaround makes the linter happy, but is quite hacky:

Add `type_hint_rename_types` to `RenameSchemaTypesVisitor` as another class attribute (parallels the [`rename_types`](https://github.com/kensho-technologies/graphql-compiler/blob/master/graphql_compiler/schema_transformation/rename_schema.py#L179) frozenset that already exists)

```
type_hint_rename_types = Union[
    EnumTypeDefinitionNode,
    InterfaceTypeDefinitionNode,
    NamedTypeNode,
    ObjectTypeDefinitionNode,
    UnionTypeDefinitionNode,
]
```

Then we can use this `Union` as the type in the function signature for `_rename_name_and_add_to_record` and replace the call [`self._rename_name_and_add_to_record(node)`](https://github.com/kensho-technologies/graphql-compiler/blob/master/graphql_compiler/schema_transformation/rename_schema.py#L268) with 

```
renamed_node = self._rename_name_and_add_to_record(
    cast(RenameSchemaTypesVisitor.type_hint_rename_types, node)
)
```

Trying to turn the existing set to a mypy `Union` doesn't work syntactically (e.g. iterating over these 5 types don't work) as far as I can tell, but I'd be very happy to be proven wrong!